app/services/instagram/test_event_service.rb

```diff
@@ -0,0 +1,79 @@
+class Instagram::TestEventService
+  def initialize(messaging)
+    @messaging = messaging
+  end
+
+  def perform
+    Rails.logger.info("Processing Instagram test webhook event, #{@messaging}")
+
+    return false unless test_webhook_event?
+
+    create_test_text
+  end
+
+  private
+
+  def test_webhook_event?
+    @messaging[:sender][:id] == '12334' && @messaging[:recipient][:id] == '23245'
+  end
+
+  def create_test_text
+    # As of now, we are using the last created instagram channel as the test channel,
+    # since we don't have any other channel for testing purpose at the time of meta approval
+    channel = Channel::Instagram.last
+
+    @inbox = ::Inbox.find_by(channel: channel)
```

Comment:
Is there a way we can find the channel by something from the payload?

Reply:
No. There is no way to get the connected channel instagram id.
![CleanShot 2025-04-11 at 15 47 07@2x](https://github.com/user-attachments/assets/ace1e71c-6d48-46ca-a69e-1be5b5bfa2eb)